crates/task-impls/src/quorum_proposal/mod.rs

```diff
@@ -344,6 +344,17 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> QuorumProposalTaskState<TYPE
 
             self.latest_proposed_view = new_view;
 
+            return true;
+        } else if self.quorum_membership.leader(new_view) == self.public_key
+            && self.quorum_membership.leader(new_view + 1) == self.public_key
+        {
+            // this means we were the leader for the current view and for the next view we will be leader again
+            // there is no dependency task for the current view and we need to create one for the next view
+            // so we return true
+            debug!(
+                "Leader for connsecutive views, latest proposed view {}",
+                *self.latest_proposed_view,
+            );
```

Comment:
Wonder what other opinions on this are but to me it seems like the proposal task should just not short circuit if this function returns false, and the `create_dependency_task_if_new` fn should handle making sure we don't create multiples for the same view.  To me it looks like we already check `self.latest_proposed_view` before creating the dependency task so we probably don't need a special case in this function

Reply:
Yeah I think hard-coding for this could be a code smell. I feel like I'm not clear on the chain here. So if we're the leader for two views in a row, why does that stop us from proposing?

Reply:
yea that logic wasnt needed. thanks for pointing that out i was just trying to make sure it got to the `create_dependency_task_if_new`. but removing the return statement will be enough, even if we dont update latest proposed view, because as brendon pointed out we already have some safety checks in if there are dependency tasks for a view.